discovery/syncer.go

```diff
@@ -371,6 +371,13 @@ type GossipSyncer struct {
 	// state.
 	newChansToQuery []lnwire.ShortChannelID
 
+	// proofSentToChan is used when we already have the fully assembled
+	// proof for a channel and the peer sending us their proof has probably
+	// not received our local proof yet. So we are kind and send them our
+	// proof, but only if we haven't done so since (re)connecting. We keep
+	// track of sends with this map, so we don't send it twice.
+	proofSentToChan fn.Set[lnwire.ChannelID]
+
 	cfg gossipSyncerCfg
```

Comment:
gossip syncers are for syncing general graph gossip. The gossip syncers have nothing to do with our local channel announcement construction. So this is not where this should live imo

Reply:
note that in future we will add the ability to switch off gossip syncers (for nodes using a remote graph instance) - but then all the logic for opening our own channels should still work as normal

Reply:
also note that not all peers want to sync gossip & so wont advertise `lnwire.GossipQueriesOptional` and so we would never initialise a syncer for them in the first place